pkg/suggest/rules.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package suggest
+
+import (
+	"errors"
+	"fmt"
+	"github.com/agext/levenshtein"
+	"regexp"
+	"strings"
+)
+
+type rule struct {
+	condition      conditionFunc
+	transformation transformFunc
+}
+
+// conditionFunc is a function that returns true if the rule should be applied to the error.
+type conditionFunc func(error) bool
+
+// transformFunc is a function that defines how the error should be transformed.
+type transformFunc func(error) error
+
+// ruleInterface represents a suggestion rule.
+type ruleInterface interface {
+	apply(error) error
+}
+
+// newRule creates a new ruleInterface instance.
+func newRule(condition conditionFunc, transform transformFunc) ruleInterface {
+	return &rule{
+		condition:      condition,
+		transformation: transform,
+	}
+}
+
+// apply executes the rule on the error.
+func (g *rule) apply(err error) error {
+	if g.condition(err) {
+		return g.transformation(err)
+	}
+	return err
+}
+
+// newStrReplaceRule creates a ruleInterface that finds and replaces a string in the error message
+func newStrReplaceRule(find, replace string) ruleInterface {
+	condition := func(e error) bool {
+		return strings.Contains(e.Error(), find)
+	}
+
+	transformation := func(e error) error {
+		replacedMessage := strings.ReplaceAll(e.Error(), find, replace)
+		return errors.New(replacedMessage)
+	}
+
+	return newRule(condition, transformation)
+}
+
+// newRegexRule creates a ruleInterface based on a regex pattern.
+func newRegexRule(pattern string, transform transformFunc) ruleInterface {
+	re := regexp.MustCompile(pattern)
+
+	condition := func(e error) bool {
+		return re.MatchString(e.Error())
+	}
+
+	return newRule(condition, transform)
+}
+
+// newLevenshteinRule creates a ruleInterface that matches a regex pattern, extracts a group,
+// and computes the Levenshtein distance for that group against a target string.
+func newLevenshteinRule(pattern string, target string) ruleInterface {
+	re, err := regexp.Compile("(.*?)" + pattern + "(.*)") // Capture everything before and after the pattern
+
+	threshold := 3
+	if len(target) <= threshold {
+		threshold = 1
+	}
+
+	condition := func(e error) bool {
+		if err != nil {
+			return false
+		}
+		matchingErrors := re.FindAllStringSubmatch(e.Error(), -1)
+		for _, matchingError := range matchingErrors {
+			distance := levenshtein.Distance(target, matchingError[2], nil)
```

Comment:
> so if distance is < 3 (can we use a const to add more context to this value?)

✅  done!

> should we calculate this rule using go routines and first matching return it? in order to not calculate distance for every key sequently

this could be a performance optimization, not sure if it's worth it, to be sure I've tried comparing execution time between okteto stable and this branch, but I cannot see any difference. Perhaps if we had hundreds of rules like this it would be beneficial to refactor. WHYT? 

Happy to give it a try together if you like